src/random-seed/random-seed.c

```diff
@@ -363,13 +363,13 @@ static int run(int argc, char *argv[]) {
 
                 /* Let's make this whole job asynchronous, i.e. let's make ourselves a barrier for
                  * proper initialization of the random pool. */
-                k = getrandom(buf, buf_size, GRND_NONBLOCK);
-                if (k < 0 && errno == EAGAIN && synchronous) {
+                k = RET_NERRNO(getrandom(buf, buf_size, GRND_NONBLOCK));
```

Comment:
this is not correct. RET_NERRNO() operates on `int` (i.e. 32bit), but getrandom() returns `ssize_t` (i.e. 64bit on modern archs)

Reply:
Indeed but the only possible negative value that can be returned by `getrandom()` should be -1, which fit an int (whatever the number of bits used to represent an int) so this should be ok, no ?